movai_core_shared/core/zmq/zmq_client.py

```diff
@@ -98,53 +172,77 @@ async def send(self, msg: dict, use_lock: bool = False) -> None:
         Asynchrounously send the message to the server.
 
         Args:
-            data (bytes): the msg representation
+            msg (dict): The message to send.
+            use_lock (bool): Whether to use the lock.
+        Returns:
+            None
         """
-        if use_lock:
-            self._init_lock()
         try:
             data = create_msg(msg)
             if use_lock and self._lock:
                 async with self._lock:
                     await self._socket.send(data)
             else:
                 await self._socket.send(data)
-        except asyncio.CancelledError:
-            # This is a normal exception that is raised when the task is CancelledError
+        except asyncio.CancelledError as exc:
+            # This is a normal exception that is raised when the task is cancelled
             self._socket.close()
+            raise exc
+        except TypeError as exc:
+            self._logger.error(
+                f"{self.__class__.__name__} failed to send message. "
+                + f"{self._addr} got type exception: {exc} "
+            )
+            raise exc
```

Comment:
we catch the exception just to log and re-raise it?

Reply:
yes, this error is not critical, message not sent because of maybe wrong structure (pydantic checks) so I jsut wanted to instrument